packages/jasmine-extensions/src/retry-failed-tests.ts

```diff
@@ -1,39 +1,41 @@
-// From: https://github.com/jasmine/jasmine/issues/960#issuecomment-2349768674
+// Modified From: https://github.com/jasmine/jasmine/issues/960#issuecomment-2349768674
 /* eslint-disable */
 export function retryFailedTests(
     retries: number,
     millisecondsBetweenRetries: number,
 ): void {
+    // Handles both styles of async (Promises and done()) and returns a
+    // Promise.  Wraps synchronous fns in a Promise, too.
+    const run = (that: any, fn: Function) => {
+        if (fn.length == 0) {
+            return fn.call(that);
+        }
+        return new Promise(resolve => {
+            fn.call(that, resolve);
+        });
+    };
+
     const setTimeout = globalThis.setTimeout;
-    // @ts-expect-error sdf
+    // @ts-expect-error Global Jasmine Spec type undefined
     const originalSpecConstructor = jasmine.Spec;
-    // @ts-expect-error sdf
+    // @ts-expect-error Global Jasmine Spec type undefined
     jasmine.Spec = function retrySpec(attrs: any): jasmine.Spec {
         const spec = new originalSpecConstructor(attrs);
         const originalTestFn = spec.queueableFn.fn;
-
-        // Handles both styles of async testing (Promises and done()) and returns a
-        // Promise.  Wraps synchronous tests in a Promise, too.
-        const runOriginalTest = () => {
-            if (originalTestFn.length == 0) {
-                return originalTestFn();
-            }
-            return new Promise(resolve => {
-                originalTestFn(resolve);
-            });
-        };
+        const beforeAfterFns = spec.beforeAndAfterFns();
```

Comment:
Wonder if this includes beforeAll and aferAll... because we don't want those to run here, only the each fns

Reply:
Included spys for those in the tests